src/setuptools_betterproto/_command.py

```diff
@@ -59,53 +67,99 @@ class CompileProto(setuptools.Command):
+            proto_path=self.proto_path,
+            proto_glob=self.proto_glob,
+            include_paths=self.include_paths,
+            out_path=self.out_path,
+        )
+
 
-    def _expand_paths(
-        self, proto_path: pathlib.Path, proto_glob: str
-    ) -> Iterator[pathlib.Path]:
-        """Expand the paths to the proto files."""
-        return (p.relative_to(proto_path) for p in proto_path.rglob(proto_glob))
+class CompileBetterproto(BaseProtoCommand):
+    """A command to compile the protobuf files."""
 
+    @override
     def run(self) -> None:
-        """Compile the Python protobuf files."""
-        include_paths = list(map(pathlib.Path, self.include_paths.split(",")))
-        proto_path = pathlib.Path(self.proto_path)
-        proto_files = list(self._expand_paths(proto_path, self.proto_glob))
+        """Compile the protobuf files to Python."""
+        proto_files = self.config.expanded_proto_files
 
         if not proto_files:
-            print(
-                f"No proto files found in {self.proto_path}/**/{self.proto_glob}/, "
-                "skipping compilation of proto files."
+            _logger.warning(
+                "No proto files found in %s with glob "
+                "%s, skipping compilation of proto files.",
+                self.config.proto_path,
+                self.config.proto_glob,
             )
             return
 
         protoc_cmd = [
             sys.executable,
             "-m",
             "grpc_tools.protoc",
-            *(f"-I{p}" for p in [self.proto_path, *include_paths]),
-            f"--python_betterproto_out={self.out_path}",
-            *map(str, proto_files),
+            *(f"-I{p}" for p in [self.config.proto_path, *self.config.include_paths]),
+            f"--python_betterproto_out={self.config.out_path}",
+            *proto_files,
         ]
 
-        print(f"Compiling proto files via: {' '.join(protoc_cmd)}")
+        _logger.info("compiling proto files via: %s", " ".join(protoc_cmd))
         subprocess.run(protoc_cmd, check=True)
 
 
-# This adds the build_betterproto command to the build sub-command.
-# The name of the command is mapped to the class name in the pyproject.toml file,
-# in the [project.entry-points.distutils.commands] section.
-# The None value is an optional function that can be used to determine if the
-# sub-command should be executed or not.
-_build_command.build.sub_commands.insert(0, ("build_betterproto", None))
+class AddProtoFiles(BaseProtoCommand):
+    """A command to add the proto files to the source distribution."""
+
+    def run(self) -> None:
+        """Copy the proto files to the source distribution."""
+        proto_files = self.config.expanded_proto_files
+        include_files = self.config.expanded_include_files
+
+        if include_files and not proto_files:
+            _logger.warning(
+                "Some proto files (%s) were found in the `include_paths` (%s), but "
```

Comment:
```suggestion
                "Some files (%s) were found in the `include_paths` (%s), but "
```

Reply:
I think the current wording is more clear. We found proto files, they are just in the `include_paths` instead of the `proto_path`, but they are still proto files IMHO.